src/ccmain/osdetect.cpp

```diff
@@ -373,9 +373,8 @@ bool OrientationDetector::detect_blob(BLOB_CHOICE_LIST *scores) {
         for (choice_it.mark_cycle_pt(); !choice_it.cycled_list() && choice == nullptr;
              choice_it.forward()) {
           int choice_script = choice_it.data()->script_id();
-          unsigned s = 0;
-          for (s = 0; s < allowed_scripts_->size(); ++s) {
-            if ((*allowed_scripts_)[s] == choice_script) {
+          for (auto script : *allowed_scripts_) {
```

Comment:
Can be `if (std::ranges::find(*allowed_scripts_, choice_script) != allowed_scripts_->end()) ...`